lib/jxl/decode_to_jpeg.h

```diff
@@ -18,12 +18,17 @@
 
 #include "jxl/decode.h"
 #include "lib/jxl/base/status.h"
+#include "lib/jxl/common.h"  // JPEGXL_ENABLE_TRANSCODE_JPEG
 #include "lib/jxl/image_bundle.h"
 #include "lib/jxl/jpeg/dec_jpeg_data.h"
+#if JPEGXL_ENABLE_TRANSCODE_JPEG
 #include "lib/jxl/jpeg/dec_jpeg_data_writer.h"
```

Comment:
It might be simpler to do the implementation of all the functions in the .cc file, and this inclusion in the cc file as well, so only in the cc file you have the macro toggle and this implementation related header inclusion.

But up to you

Reply:
I definitively want the disabled-case version to be inlined into decode.cc so all the code behind `if (functionThatReturnsFalse())` is removed by the compiler and we save on that size. For the other ones I kept the inline decisions in decode.cc and only moved the code here.